File: flask/app/routes/ums/Menu.py
from flask import Blueprint, request
from app.services.ums.MenuService import *
from app.utils.response import success_response, error_response

menu_bp = Blueprint('menu', __name__, url_prefix='/menu')

# 添加菜单
@menu_bp.route("/add", methods=["POST"])
def add_menu():
    data = request.json
    return create_menu(data)

# 修改菜单
@menu_bp.route("/edit/<int:menu_id>", methods=["POST"])
def edit_menu(menu_id):
    data = request.json
    return update_menu(menu_id, data)

# 获取所有菜单
@menu_bp.route("/list", methods=["GET"])
def list_menu():
    page = request.args.get("page",1,type=int)
    page_size = request.args.get("page_size",6,type=int)
    level = request.args.get("level", type=int)
    window_key = request.args.get("window_key", type=str)
    return get_all_menu(page,page_size,level,window_key)

# 删除菜单
@menu_bp.route("/delete/<int:menu_id>", methods=["POST"])
def delete_menu(menu_id):
    return remove_menu(menu_id)

# 修改菜单显示状态
@menu_bp.route("/updateHidden/<int:menu_id>", methods=["POST"])
def change_menu_status(menu_id):
    hidden = request.json.get("hidden")
    return update_menu_status(menu_id, hidden)

# 以树形结构返回所有菜单
@menu_bp.route("/treeList", methods=["GET"])
def tree_list():
    window_key = request.args.get("window_key", type=str)
    return get_menu_tree(window_key)

File: vue/src/api/ums/menu.js
import request from '@/utils/request'

// 获取菜单列表
export function getMenuList(page, page_size, level,window_key) {
  return request({
    url: '/menu/list',
    method: 'get',
    params: { page, page_size , level,window_key}
  })
}

// 获取所有菜单（树形结构）
export function getMenuTree(window_key) {
  return request({
    url: '/menu/treeList',
    method: 'get',
    params: {window_key}
  })
}

// 添加菜单
export function addMenu(data) {
  return request({
    url: '/menu/add',
    method: 'post',
    data
  })
}

// 编辑菜单
export function updateMenu(menu_id, data) {
  return request({
    url: `/menu/edit/${menu_id}`,
    method: 'post',
    data
  })
}

// 删除菜单
export function deleteMenu(menu_id) {
  return request({
    url: `/menu/delete/${menu_id}`,
    method: 'post'
  })
}

// 修改菜单显示状态
export function updateMenuStatus(menu_id, hidden) {
  return request({
    url: `/menu/updateHidden/${menu_id}`,
    method: 'post',
    data: { hidden }
  })
}
